### DISCUSSION TITLE
How to setup user and pass in connectionstring using  EasyCaching.CSRedis?

### DISCUSSION THREAD

**Discussion by mendezcristian** [dotnetcore/EasyCaching#496] [Q&A]
Im working with EasyCaching.CSRedis where my server redis is configured with user and pass, but I could not found any way in order to connect to server.
Someone know if it is supported?

**Comment by catcherwong** [comment]
@mendezcristian  following https://github.com/2881099/csredis

and here is a sample:

```
127.0.0.1:6379,user=abc,password=123,defaultDatabase=13,prefix=my_
```

**Reply by mendezcristian** [reply]
you example is not working, We found the issue and we reported in https://github.com/2881099/csredis/issues/487

**Reply by catcherwong** [reply]
I have a test with the newest version of [2881099/csredis](https://github.com/2881099/csredis), and it works well.

And I notice that it support `auth user password` after v3.8.0, so we should update the version to solve this issue. /cc @Memoyu

**Reply by mendezcristian** [reply]
Hi, thanks for your response.
I disassembled the 3.8.671.0 dll and in the RedisClientPoolPolicy class you can see in the BuildConnectionString method how the problem is still there, when the connection string is built, a comma is missing in front of user= , this causes that the created connectionstring to remain in an incorrect state since the host is joined to the user, without a comma between them.
please take look at this link in order to get more detail about the issue

![Error4](https://github.com/dotnetcore/EasyCaching/assets/110913373/bd050858-c2a9-4ee6-835b-9c7d2566e5b9)

**Reply by catcherwong** [reply]
After #499 was merged

1. correct user and password, test can pass
2. invalid user and password, test can not pass.

https://github.com/dotnetcore/EasyCaching/blob/151be2b646d3dcb6dbf62736f302e1d3308936b8/test/EasyCaching.UnitTests/CachingTests/CSRedisCachingProviderTest.cs#L44-L68
